crates/trie/trie/src/forward_cursor.rs

```diff
@@ -23,18 +23,32 @@ impl<'a, K, V> ForwardInMemoryCursor<'a, K, V> {
 
 impl<K, V> ForwardInMemoryCursor<'_, K, V>
```

Comment:
since now it matters, please add a `debug_assert!(entries.is_sorted())` and change the entries type to a slice instead of ref vec